src/middlewares.js

```diff
@@ -86,21 +87,30 @@ export async function refreshUserFromIDMService(req, res, next) {
 
 export function extendJWTExpiration(req, res, next) {
   if (process.env.JWT_PRIVATE_KEY && req.user) {
-    try {
-      const jwtClaims = jwtClaimsForUser(req.user)
-      const expires = new Date(jwtClaims.exp * 1000)
-      const token = jwt.sign(jwtClaims, process.env.JWT_PRIVATE_KEY, {algorithm: 'RS512'})
-      req.lgJWT = token
-      // console.info('Extending JWT expiration.')
-      res.set('LearnersGuild-JWT', token)
-      res.cookie('lgJWT', token, Object.assign(cookieOptsJWT(req), {expires}))
-    } catch (err) {
-      console.error('Invalid JWT:', err.message ? err.message : err)
-      res.clearCookie('lgJWT', cookieOptsJWT(req))
-      req.lgJWT = null
+    if (req.user.active) {
+      try {
+        const jwtClaims = jwtClaimsForUser(req.user)
+        const expires = new Date(jwtClaims.exp * 1000)
+        const token = jwt.sign(jwtClaims, process.env.JWT_PRIVATE_KEY, {algorithm: 'RS512'})
+        req.lgJWT = token
+        // console.info('Extending JWT expiration.')
+        res.set('LearnersGuild-JWT', token)
+        res.cookie('lgJWT', token, Object.assign(cookieOptsJWT(req), {expires}))
+      } catch (err) {
+        console.error('Invalid JWT:', err.message ? err.message : err)
+        revokeJWT(req, res)
+      }
+    } else {
+      console.log(`Inactive User [${req.user.id}] (${req.user.handle}). Revoking JWT`)
+      revokeJWT(req, res)
     }
   }
   if (next) {
     next()
   }
 }
+
+function revokeJWT(req, res) {
+  res.clearCookie('lgJWT', cookieOptsJWT(req))
+  req.lgJWT = null
```

Comment:
It seems a little too magical for me for a middleware named `extendJWTExpiration` to have a side affect of changing `req.user`. I think I'd prefer it if `refreshUserFromIDMService` was responsible for nulling out the `req.user` since it seems more related to it's current responsibility.